src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/crushmap/crushmap.component.ts

```diff
@@ -39,17 +39,19 @@ export class CrushmapComponent implements OnInit, OnDestroy {
   metadata: any;
   metadataTitle: string;
   metadataKeyMap: { [key: number]: any } = {};
+  data$: Observable<object>;
 
-  constructor(private healthService: HealthService, private timerService: TimerService) {}
+  constructor(private crushRuleService: CrushRuleService, private timerService: TimerService) {}
 
   ngOnInit() {
-    this.healthService.getFullHealth().subscribe((data: any) => {
+    this.crushRuleService.getInfo().subscribe((data: any) => {
       this.loadingIndicator = false;
       this.nodes = this.abstractTreeData(data);
     });
     this.sub = this.timerService
-      .get(() => this.healthService.getFullHealth(), 5000)
+      .get(() => this.crushRuleService.getInfo(), 5000, 5000)
       .subscribe((data: any) => {
+        this.loadingIndicator = false;
         this.nodes = this.abstractTreeData(data);
       });
```

Comment:
Isn't this duplicate code (1st request, subsequent polling)? Can't we replace all this with a single call, perhaps just?:
```ts
    this.sub = this.timerService
      .get(() => this.crushRuleService.getInfo(), 5000, 5000)
      .subscribe((data: any) => {
        this.loadingIndicator = false;
        this.nodes = this.abstractTreeData(data);
      });
```

Reply:
Yes, did it with a single call.